flang/test/Lower/intrinsic-procedures.f90

```diff
@@ -70,6 +82,18 @@ subroutine anint_test(a, b)
   b = anint(a)
 end subroutine
 
+! ANY
+! CHECK-LABEL: any_test
+! CHECK-SAME: %[[arg0:.*]]: !fir.box<!fir.array<?x!fir.logical<4>>>) -> !fir.logical<4>
+logical function any_test(mask)
+  logical :: mask(:)
+! CHECK: %[[c1:.*]] = constant 1 : index
+! CHECK: %[[a1:.*]] = fir.convert %[[arg0]] : (!fir.box<!fir.array<?x!fir.logical<4>>>) -> !fir.box<none>
+! CHECK: %[[a2:.*]] = fir.convert %[[c1]] : (index) -> i32
+  any_test = any(mask)
+! CHECK:  %[[a3:.*]] = fir.call @_FortranAAny(%[[a1]], %{{.*}}, %[[a2]]) : (!fir.box<none>, !fir.ref<i8>, i32, i32) -> i1
```

Comment:
There are 4 arguments, but the regular expression appears to only be looking for 3. (Two get matched with the single `%{{.*}},` in the middle.) That's a bit confusing at first glance.

Reply:
The runtime routine's arguments are:

const Descriptor &x, const char *source, int line, int dim

The two arguments that are being "wild carded" are the source info arguments that we usually "wild card" from what I've seen in other tests. The first argument is the mask argument and the last argument is the optional dim argument.

Reply:
Let me know if there's a better way to write this.

Reply:
I was thinking something like `%{{.*}}, %{{.*}},` would make it clearer that you expect 2 distinct arguments.